src/SFA.DAS.AANHub.Application/CalendarEvents/Queries/GetCalendarEventsQueryHandler.cs

```diff
@@ -0,0 +1,37 @@
+﻿using MediatR;
+using SFA.DAS.AANHub.Application.Mediatr.Responses;
+using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+
+namespace SFA.DAS.AANHub.Application.CalendarEvents.Queries;
+
+public class GetCalendarEventsQueryHandler : IRequestHandler<GetCalendarEventsQuery, ValidatedResponse<GetCalendarEventsQueryResult>>
+{
+    private readonly ICalendarEventsReadRepository _calendarEventsReadRepository;
+
+    public GetCalendarEventsQueryHandler(ICalendarEventsReadRepository calendarEventsReadRepository)
+    {
+        _calendarEventsReadRepository = calendarEventsReadRepository;
+    }
+
+    public async Task<ValidatedResponse<GetCalendarEventsQueryResult>> Handle(GetCalendarEventsQuery request, CancellationToken cancellationToken)
+    {
+        var pageSize = Domain.Common.Constants.CalendarEvents.PageSize;
+        var page = 1;
+        var response =
+            await _calendarEventsReadRepository.GetCalendarEvents(request.RequestedByMemberId);
+
+
+        var orderedResponse = response.OrderBy(p => p.Start);
```

Comment:
Why not order by in the query?

Reply:
because orderby will become variable (future stories make is by 'distance' or 'date')and I felt this was more readable.  We're getting all the data back anyway before it gets paginated, so felt the ordering logic felt appropriate for this layer rather than the database layer.

Reply:
I could embed it in the database layer though, if you think it's better?